### PR TITLE
DB changes proposed during meeting

### PR DESCRIPTION
**Pull request by GrantDouglas**
# Summary

This PR consists of the DB changes that were proposed during the meeting today. These changes include the token table, the renaming of fields within the recipe table, and removes duplicate entries upon initial food seeding.

# Acceptance Criteria

> What needs to be done and tested for this PR to be merged

- [x] Account table is created properly with proper fields
- [x] Recipe table has the correct fields
- [x] Food table does not create duplicates on initial seeding
- [x] Food table allows entries with similar titles to be entered upon usage

### CODE REVIEW

File: db/init.sql [hunk 1]
```diff
@@ -300,3 +326,20 @@ ALTER TABLE food DROP COLUMN usda_id;
 ALTER TABLE nutrition DROP COLUMN usda_id;
 DROP SCHEMA usda CASCADE;
 DROP EXTENSION tablefunc;
+
+UPDATE food
```
Comment (jkwill87) [inline db/init.sql]: I'm not sure that we want these here for the following reasons:

1. You are referencing `recipe_id`s that don't yet exists in the recipe table
2. Food IDs aren't guaranteed to be the same after initialisation, e.g. if changes are ever made to the tables, like the one here where only unique ingredients are added, or if the USDA csv is updated
3. Even if we did want to create dummy recipes we might want to split that into a separate .sql file

File: db/init.sql [hunk 2]
```diff
@@ -142,6 +166,7 @@ CREATE TABLE usda.products (
   date_modified TEXT,
```
Comment (jkwill87) [inline db/init.sql]: should be `TIMESTAMP`

File: db/init.sql [hunk 3]
```diff
@@ -142,6 +166,7 @@ CREATE TABLE usda.products (
...
   date_available TEXT,
```
Comment (jkwill87) [inline db/init.sql]: should be TIMESTAMP

File: db/init.sql [hunk 4]
```diff
@@ -163,7 +188,8 @@ ALTER TABLE usda.products
   DROP COLUMN ingredients_english;
 INSERT INTO food (usda_id, title)
 SELECT ndb_number, initcap(long_name)
-FROM usda.products;
+FROM usda.products
+ON CONFLICT ON CONSTRAINT title_const DO NOTHING;
```
Comment (jkwill87) [inline db/init.sql]: :+1:

File: db/init.sql [hunk 5]
```diff
@@ -142,6 +166,7 @@ CREATE TABLE usda.products (
   date_modified TEXT,
   date_available TEXT,
   ingredients_english TEXT
```
Comment (jkwill87) [inline db/init.sql]: This should probably be `food_ids INTEGER[]`

File: db/init.sql [hunk 6]
```diff
@@ -109,8 +133,8 @@ CREATE TABLE food (
   recipe_id INTEGER REFERENCES recipe UNIQUE,
   time_created TIMESTAMP DEFAULT now() NOT NULL,
   time_updated TIMESTAMP DEFAULT now() NOT NULL,
-  view_count INTEGER DEFAULT 0,
-  star_count INTEGER DEFAULT 0
+  CONSTRAINT title_const UNIQUE(title)
+
 );
 
 -- Meal
```
Comment (jkwill87) [inline db/init.sql]: From what we discussed today we will need to have this table defined, although it can be done in a separate PR.

Comment (GrantDouglas) [reply db/init.sql]: yea was going to do some more planning on those new tables. Just wanted the ones from last milestone done.

File: db/init.sql (hunk 1 above)
Comment (GrantDouglas) [reply db/init.sql]: Yea these can be (re)moved if necessary. Only had them there for Filips convenience before the demo. They won't be needed when we get an easier way to create recipes finished

**Pull request merged**